python/serve.py

```
@@ -38,6 +97,66 @@ def decode(token_ids: list[int]) -> str:
+    Processes the prompt once (prefill), then generates one token at a time
+    using cached K/V projections. ~100-500x faster per decode step vs full forward.
+    """
+    seq = list(prompt_tokens)
+    vocab = cfg.vocab_size
+    seq_len = cfg.seq_len
+
+    owns_conductor = False
+    if conductor is None:
+        conductor = nl_hecate.Conductor(
+            cfg.k, list(cfg.chunk_sizes) if hasattr(cfg, 'chunk_sizes') else [1] * cfg.k)
+        owns_conductor = True
```

Ruff (0.15.1)

[error] Local variable `owns_conductor` is assigned to but never used

Remove assignment to unused variable `owns_conductor`

(F841)